process/dns_v2.go

```diff
@@ -330,11 +330,13 @@ func getDNSNameAsByteSliceByOffset(buf []byte, offset int) (stringasbyteslice []
 	if offset+int(namelen) > len(buf) {
 		return nil, fmt.Errorf("offset out of range [%d:%d] > %d", offset, offset+int(namelen), len(buf))
 	}
-
+	// For when the domain name is an empty string
+	if namelen == 0 {
+		return nil, fmt.Errorf("illegal domain namelen of 0")
+	}
 	if offset+int(namelen) <= offset {
-		return nil, fmt.Errorf("illegal namelen")
+		return nil, fmt.Errorf("integer overflow: (%d+%d) <= %d", offset, int(namelen), offset)
 	}
```

Comment:
I think you could just have `if int(namelen) < 0`, but since it was like that before I'd understand if you want to leave it that way.

I'd change the error message to be clearer (if you see `integer overflow` without any context you have no idea where it comes from), eg. something like `domain name length is too big`.